core/index/src/main/java/mil/nga/giat/geowave/core/index/ByteArrayRange.java

```diff
@@ -108,26 +109,28 @@ public int compareTo(
 				other.getEnd());
 	}
 
-	public static final void mergeIntersections(
+	public static final List<ByteArrayRange> mergeIntersections(
 			List<ByteArrayRange> ranges,
 			int maxRanges ) {
 		// sort order so the first range can consume following ranges
 		Collections.<ByteArrayRange> sort(ranges);
-		// merge in place
-		for (int i = 0; i < ranges.size(); i++) {
+		final List<ByteArrayRange> result = new ArrayList<ByteArrayRange>();
+		for (int i = 0; i < ranges.size(); ) {
 			ByteArrayRange r1 = ranges.get(i);
-			for (int j = i + 1; j < ranges.size(); j++) {
+			int j = i + 1;
+			for (; j < ranges.size(); j++) {
 				final ByteArrayRange r2 = ranges.get(j);
 				if (r1.intersects(r2)) {
 					r1 = r1.merge(r2);
-					ranges.remove(j);
-					j--;
-					ranges.set(
-							i,
-							r1);
+				}
+				else {
+					break;
 				}
 			}
+			i = j;
```

Comment:
got it, sorry for the confusion...`a` will merge with `b`, then `c`, then `d` in my scenario so `b` and `c` will never need to be merged directly together, ignore my comment